dlt/destinations/weaviate/weaviate_client.py

```diff
@@ -0,0 +1,540 @@
+            ),
+            consistency_level=weaviate.ConsistencyLevel[
+                self.client_config.batch_consistency
+            ],
+            num_workers=self.client_config.batch_workers,
+            callback=check_batch_result,
+        ) as batch:
+            for line in f:
+                data = json.loads(line)
+                # make complex to strings
+                for key in self.complex_indices:
+                    if key in data:
+                        data[key] = json.dumps(data[key])
+                for key in self.date_indices:
+                    if key in data:
+                        data[key] = str(ensure_pendulum_datetime(data[key]))
+                if self.unique_identifiers:
+                    uuid = self.generate_uuid(
+                        data, self.unique_identifiers, self.class_name
+                    )
+                else:
+                    uuid = None
+
+                batch.add_data_object(data, self.class_name, uuid=uuid)
+
+    def list_unique_identifiers(self, table_schema: TTableSchema) -> Sequence[str]:
+        if table_schema.get("write_disposition") == "merge":
+            primary_keys = get_columns_names_with_prop(table_schema, "primary_key")
+            if primary_keys:
+                return primary_keys
+        return get_columns_names_with_prop(table_schema, "unique")
+
+    def generate_uuid(
+        self, data: Dict[str, Any], unique_identifiers: Sequence[str], class_name: str
+    ) -> str:
+        data_id = "_".join([str(data[key]) for key in unique_identifiers])
+        return generate_uuid5(data_id, class_name)  # type: ignore
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class WeaviateClient(JobClientBase):
+    """Weaviate client implementation."""
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: WeaviateClientConfiguration) -> None:
+        super().__init__(schema, config)
+        self.config: WeaviateClientConfiguration = config
+        self.db_client = self.create_db_client(config)
+
+        self._vectorizer_config = {
+            "vectorizer": config.vectorizer,
+            "moduleConfig": config.module_config,
+        }
+
+    @staticmethod
+    def create_db_client(config: WeaviateClientConfiguration) -> weaviate.Client:
+        return weaviate.Client(
+            url=config.credentials.url,
+            auth_client_secret=weaviate.AuthApiKey(api_key=config.credentials.api_key),
+            additional_headers=config.credentials.additional_headers,
+        )
+
+    def make_full_name(self, table_name: str) -> str:
+        """Make a full Weaviate class name from a table name by prepending
+        the dataset name if it exists.
+        """
+        return (
+            f"{self.config.dataset_name}_{table_name}"
```

Comment:
here's the new thing:
use self.config.normalize_dataset_name(schema)
to get the right dataset name.

also I'm not sure we should use "_" as separator. I'd say not - naming convention does not allow for `_` in table names

Reply:
Thanks! That could be the source of the error in the tests. I'd love to use some kind of separator as it makes it easier to read the names if the prefixed classes